### DISCUSSION TITLE
The map is misleading and doesn't represent the game world

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#10684] [Unstable feedback]
This is the current in-game map
![image](https://user-images.githubusercontent.com/73229309/206239301-6f664761-790c-4818-a672-22eeb497493d.png)

**It's safe to say that most players believe:**

- You're going sideways the entire campaign; the map goes left to right, and the level end is at the same depth as the level start
- Radiation is also going sideways (it clearly is on the map)

Traversal is extremely "flat" and horizontal in level generation:
You end up at the same depth as the start, only to be teleported at a lower depth when leaving an outpost

![image](https://user-images.githubusercontent.com/73229309/206752545-7035e13e-3741-4167-b42f-48743ba6f79f.png)

Upon reaching the second biome the depth becomes wildly inconsistent:
![image](https://user-images.githubusercontent.com/73229309/206762711-28ee24c2-40fc-4222-b9a9-9a74f5991a14.png)

Only the depth meter actually indicates going deeper

This is the "official" biomes map and how you're traversing it
![image](https://user-images.githubusercontent.com/73229309/206240535-aeb0d753-158c-49bd-ad6f-03c8cc715723.png)

And the lore
![image](https://user-images.githubusercontent.com/73229309/206240662-749d023c-8310-4106-9885-b5416424dcc3.png)
`Due to the ever-increasing solar flare activity, it is necessary for us to go deeper, but many are concerned about the mysterious “gravity well” our colonies sit on top of but that we still don’t quite understand.`

# What the world actually looks like 

To my understanding, there's a gravity well (a cylinder?) going all the way to the surface, all of the colonies are established within the perimeter
![image](https://user-images.githubusercontent.com/73229309/206244587-b6791b32-101a-49dd-8b3b-266f2a0e5e8d.png)

For practical reasons you're either traversing diagonally from an edge, or in a sort of downward spiral pattern around the eye.

This is how the map should look like:
![image](https://user-images.githubusercontent.com/73229309/206245284-3772a3ef-f438-4a0b-a457-2c383c2174d1.png)

Generating locations and outposts along a diagonal while radiation is going downwards is a proper representation of how you're actually traversing Europa. The current map doesn't display that at all.

**Comment by ubertpendragon** [comment]
The map generator can manage a constant "downward slope" of depth in its parameters: here is a 900m drop
![image](https://user-images.githubusercontent.com/73229309/218117204-da0cd841-4b22-4d94-a87f-e219daf52dc1.png)

**Comment by ubertpendragon** [comment]
Taken from Discord:

1. End hole and looping around the level if you fall into the abyss is unintuitive
2. Falling deep on the side of the mountain and recovering is very slow
3. Crush depth is non-existent because pressure isn't relative to the level itself (you can "upgrade" pressure out of the game)
3.5. Abyssals aren't deadly even if they send you down into the abyss
4. The U-shaped levels make you end a level at the exact same depth you started them (and teleport you X00m deeper on the next cave)

How levels are generated, with abyss mountain and hole to the end for going outside the map
![image](https://user-images.githubusercontent.com/73229309/220963248-a5c4dd1b-2161-4c8e-8af9-f19af02549a7.png)

Potential solution: abyss "pit" instead of mountain, with hole on the right side
This "traps" the submarine inside the level and makes navigation in the abyss much simpler
![image](https://user-images.githubusercontent.com/73229309/220963577-177b10c4-4af3-4e3b-8e8b-0ef311ae46a9.png)

Changes to pressure/crush depth:
Instead of a linear depth value across the entire campaign which you are incentivized to out-upgrade (and ends up being a non-factor), make crush depth relative to each level by splitting the map in 3 layers of pressure:

Cave Layer
Abyss Layer (where the islands are)
Crush layer (if your sub falls down here you die)

![image](https://user-images.githubusercontent.com/73229309/220964305-8a1c1c84-11c6-424e-8da4-2188786e79cc.png)

**Comment by girlyguppy** [comment]
so true bestie

**Comment by Kosba2** [comment]
A lot of thought went into this and that's pretty cool
